libs/utils/src/tallies.ts

```diff
@@ -33,3 +40,31 @@ export function combineContestTallies(
     },
   };
 }
+
+export function getSubTalliesByPartyAndPrecinct({
+  election,
+  tally,
+  precinctSelection,
+}: {
+  election: Election;
+  tally: FullElectionTally;
+  precinctSelection?: PrecinctSelection;
+}): Map<string, Tally> {
+  const newSubTallies = new Map();
+  const precinctIdList = precinctSelection
+    ? precinctSelection.kind === 'AllPrecincts'
+      ? election.precincts.map(({ id }) => id)
+      : [precinctSelection.precinctId]
+    : [undefined];
```

Comment:
Yeah, it's because `undefined` maps to "All Precincts" in how we are keying the compressed tallies on the card tally. It's really not ideal - and we should type this data differently - but I want to leave it now because the possibility that it's only an all precincts tally is based on the fact that the "by precinct" data might not fit on the card. https://github.com/votingworks/vxsuite/pull/2730#discussion_r1001348329. I will add a comment in.